packages/clerk-js/src/testUtils.ts

```diff
@@ -11,6 +11,43 @@ const render = (ui: React.ReactElement, options?: RenderOptions) => {
   return { ..._render(ui, { ...options }), userEvent };
 };
 
+/**
+ * Helper method to mock a native runtime environment for specific test cases, currently targeted at React Native.
+ * Makes some assumptions about our runtime detection utilities in `packages/clerk-js/src/utils/runtime.ts`.
+ *
+ * Usage:
+ *
+ * ```js
+ * mockNativeRuntime(() => {
+ *  // test cases
+ *  it('simulates native', () => {
+ *    expect(typeof document).toBe('undefined');
+ *  });
+ * });
+ * ```
+ */
+export const mockNativeRuntime = (fn: () => void) => {
```

Comment:
Haven't seen this pattern a lot, but I really like the approach here :)

Reply:
i haven't either 😅 . I would expect the `mockNativeRuntime` to run inside each test case instead of this. I am not a huge fun of having a test helper containing `beforeAll/beforeEach/afterAll/afterEach` and `describe` in it but it is readable and it makes sense, that's why i would suggest we try this pattern and see how it goes.

Reply:
Happy to adjust! I started with the approach you have suggested here, but switched to this one when thinking about needing to add error handling and such to ensure the `mockRestore` methods were called.